smarty/sampler.py

```diff
@@ -129,41 +144,16 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
                 self.atomtypes = [(smarts, typename)] + self.atomtypes
```

Comment:
@camizanette The issue in this part of the code is independent of the input file. We should not assume that users are using our input files. So I voiced two completely separate concerns:

1. When base types are added to initial types I think they should remain in the same order and be put at the top of the list. 

2. Best practices should have base and initial types that are listed from most to least general, though technically users could do whatever they want. 

I think the best fix to this code would be to replace lines 140-144 with:
```
# Add any base types not already there to the initial types
initial_smarts = [ smarts for (smarts, name) in self.atomtypes ]
missing_basetypes = list()
for [smarts, typename] in self.basetypes:
    if smarts not in initial_smarts:
         missing_basetypes.append( (smarts, typename) )
self.atomtypes = missing_basetypes + self.atomtypes
```
This would keep any added basetypes at the top and in the same order they were in in the original input.